packages/@aws-cdk/aws-codestarnotifications/lib/notification-rule.ts

```diff
@@ -0,0 +1,237 @@
+   */
+  ENABLED = 'ENABLED',
+}
+
+/**
+ * Standard set of options for `notifyOnXxx` codestar notification handler on construct
+ */
+export interface NotifyOptions {
+  /**
+   * The name for the notification rule.
+   * Notification rule names must be unique in your AWS account.
+   *
+   * @default - generated from the `id`
+   */
+  readonly notificationRuleName?: string;
+
+  /**
+   * The target to register for the notification destination.
+   *
+   * @default - No target is added to the rule. Use `addTarget()` to add a target.
+   */
+  readonly target?: INotificationRuleTarget;
+
+  /**
+   * The status of the notification rule.
+   * If the enabled is set to DISABLED, notifications aren't sent for the notification rule.
+   *
+   * @default true
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * The level of detail to include in the notifications for this resource.
+   * BASIC will include only the contents of the event as it would appear in AWS CloudWatch.
+   * FULL will include any supplemental information provided by AWS CodeStar Notifications and/or the service for the resource for which the notification is created.
+   *
+   * @default DetailType.FULL
+   */
+  readonly detailType?: DetailType;
+}
+
+/**
+ * Standard set of options for `notifyOn` codestar notification handler on construct
+ */
+export interface NotifyOnEventOptions extends NotifyOptions {
+  /**
+   * A list of event types associated with this notification rule.
+   * For a complete list of event types and IDs, see Notification concepts in the Developer Tools Console User Guide.
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#concepts-api
+   *
+   * You must specify this property (either via props or via `addEvents`)
+   *
+   * @default - No events.
+   */
+  readonly events?: string[];
+}
+
+/**
+ * Properties for a new notification rule
+ */
+export interface NotificationRuleProps extends NotifyOnEventOptions {
+  /**
+   * The Amazon Resource Name (ARN) of the resource to associate with the notification rule.
+   * Currently, Supported sources include pipelines in AWS CodePipeline and build projects in AWS CodeBuild in this L2 constructor.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html#cfn-codestarnotifications-notificationrule-resource
+   */
+  readonly source: INotificationRuleSource;
+}
+
+/**
+ * Represents a notification rule
+ */
+export interface INotificationRule extends IResource {
+
+  /**
+   * The ARN of the notification rule (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   *
+   * @attribute
+   */
+  readonly notificationRuleArn: string;
+
+  /**
+   * Adds target to notification rule
+   *
+   * @param target The SNS topic or AWS Chatbot Slack target
+   * @returns boolean - return true if it had any effect
+   */
+  addTarget(target?: INotificationRuleTarget): boolean;
```

Comment:
`target` should be required here:

```suggestion
  addTarget(target: INotificationRuleTarget): boolean;
```